src/main/java/ru/mtsbank/fintech/animal_repository/AnimalRepositoryImpl.java

```diff
@@ -9,12 +9,13 @@
 import javax.annotation.PostConstruct;
 import java.time.LocalDate;
 import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.stream.Collectors;
 
 @Service
 public class AnimalRepositoryImpl implements AnimalRepository {
 
-    private Map<String, List<AbstractAnimal>> animalMap;
+    private ConcurrentHashMap<String, List<AbstractAnimal>> animalMap;
```

Comment:
```suggestion
    private Map<String, List<AbstractAnimal>> animalMap;
```
смотри коменты к `CreateAnimalServiceImpl`